spotbugs/src/main/java/edu/umd/cs/findbugs/sarif/BugCollectionAnalyser.java

```diff
@@ -0,0 +1,59 @@
+package edu.umd.cs.findbugs.sarif;
+
+import edu.umd.cs.findbugs.BugCollection;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugPattern;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.json.JSONArray;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+class BugCollectionAnalyser {
+    @NonNull
+    private final List<Rule> rules = new ArrayList<>();
+    @NonNull
+    private final List<Result> results = new ArrayList<>();
+    @NonNull
+    private final Map<String, Integer> typeToIndex = new HashMap<>();
+
+    BugCollectionAnalyser(@NonNull BugCollection bugCollection) {
+        bugCollection.forEach(bug -> {
+            String type = bug.getType();
+            int index = typeToIndex.computeIfAbsent(type, (t) -> processRule(bug.getBugPattern()));
+
+            processResult(index, bug);
+        });
+    }
+
+    JSONArray getRules() {
+        JSONArray array = new JSONArray();
```

Comment:
There is a JSONArray constructor that takes a collection directly; why not use it?